corehq/apps/accounting/models.py

```diff
@@ -1822,6 +1833,15 @@ def text_template(self):
         else:
             return self.INVOICE_TEXT_TEMPLATE
 
+    @property
+    def should_send_email(self):
+        subscription = self.invoice.subscription
+        autogenerate = (subscription.auto_generate_credits and not self.invoice.balance)
```

Comment:
(nonblocking) this variable name could be clearer (it took me some time to figure out what was going on here)